app/models/user.rb

```diff
@@ -80,4 +80,16 @@ def remove_pregnancy(pregnancy)
     pregnancies.delete pregnancy
     reload
   end
+
+  def clear_call_list
+    pregnancies.each do |p|
+      pregnancies.delete(p) if recently_reached?(p)
+    end
+  end
+
+  private
+
+  def recently_reached?(preg)
```

Comment:
what do you think about renaming this to `recently_reached_by_user`, for clarity's sake? Thoughts?